server/service/client.go

```diff
@@ -384,7 +392,7 @@ func (c *Client) ApplyGroup(
 		macosCustomSettings := extractAppCfgMacOSCustomSettings(specs.AppConfig)
 		allCustomSettings := append(macosCustomSettings, windowsCustomSettings...)
 
-		if len(allCustomSettings) > 0 {
+		if (windowsCustomSettings != nil && macosCustomSettings != nil) || len(allCustomSettings) > 0 {
```

Comment:
Not sure I understand why both must be non-nil? Might be a typo and meant to be an `or` (but then the other condition ensures that at least one is non-nil already?)

Reply:
I think I understand the reason (updated the code comment above), but still not sure how it should behave if a) `len(allCustomSettings) == 0` and b) only one of `windowsCustomSettings` and `macosCustomSettings` is `non-nil`?

Reply:
@roperzh do we need to address this before merging?